### ISSUE TITLE
WslRegisterDistribution failed with error: 0x8007023e

### ISSUE THREAD

**Issue by pszlaga** [microsoft/WSL#10507]
### Windows Version

Microsoft Windows [Version 10.0.19045.3324]

### WSL Version

1.2.5.0

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.15.90.1

### Distro Version

Ubuntu 20.04

### Repro Steps

wsl -d Ubuntu-20.04

### Expected Behavior

It will launch Ubuntu 20.04.

### Actual Behavior

{Błąd aplikacji} W aplikacji wystąpił wyjątek %s (0x
Error code: Wsl/Service/CreateInstance/CreateVm/ConfigureNetworking/0x8007023e

### Diagnostic Logs

[WslLogs-2023-09-20_00-02-31.zip](https://github.com/microsoft/WSL/files/12666935/WslLogs-2023-09-20_00-02-31.zip)

**Comment by pmartincic** [comment]
I don't see anything in the traces. Did you let the log collection run while launching wsl?

**Comment by pszlaga** [comment]
> I don't see anything in the traces. Did you let the log collection run while launching wsl?

https://aka.ms/AAmqjfp

**Comment by pmartincic** [comment]
@pszlaga, what I need is for the log collection program to run while you run commands on/inside wsl. If you immediately exit the log collection, I don't get to see anything.

**Comment by pszlaga** [comment]
Oh, i'm sorry, my bad.

Now you have new logs in attachment

[WslLogs-2023-09-20_01-07-59.zip](https://github.com/microsoft/WSL/files/12667173/WslLogs-2023-09-20_01-07-59.zip)

**Comment by pmartincic** [comment]
@keith-horton, can you triage or delegate triage? This is pre mirrored and shows errors on `hns-create*`

**Comment by keith-horton** [comment]
It's failing to create the switch (network). @pszlaga , can you collect a trace using this WPR script https://github.com/microsoft/WSL/blob/master/diagnostics/wsl_networking.wprp? We need some network-specific traces:

wpr -start wsl_networking.wprp -filemode
<repro>
wpr.exe -stop wsl_repro.etl

**Comment by pszlaga** [comment]
[wsl_repro.zip](https://github.com/microsoft/WSL/files/12678556/wsl_repro.zip)

**Comment by pszlaga** [comment]
Hello, do you have any solutions?

**Comment by keith-horton** [comment]
Sorry - I reviewed the traces and opened an issue with the Hyper-V VSwitch team. Their traces don't show the source of the error - only that it's failing trying to create that NAT network.

Given this I'm trying to find a way to unblock you. This continue to repro after a reboot?
If so, what optional-features do you have installed? I would try to remove Hyper-V if possible (if it doesn't break you or lose hard-to-recover configurations).
These Windows Features all install vSwitch: Hyper-V, Microsoft Defender Application Guard, Virtual Machine Platform, Windows Sandbox.

WSL requires Virtual Machine Platform - though you can have others installed as well.

If it doesn't risk you losing data, and just a reboot won't fix this, I would remove the features above, reboot, then add what you need back.

I'm sorry I don't have a better work-around available.

**Comment by pszlaga** [comment]
I have everything disabled except:

- Windows subsystem for linux
- Virtual machine platform

I uninstalled the above and reinstalled it.

I also removed WSL and reinstalled it via PowerShell.

I used the commands:

netcfg -d
winsock reset

Unfortunately, nothing helps.

Tell me, am I able to recover files from a vhdx disk when the system was Ubuntu 20.04?

wsl export also returns the same error as when I want to start the machine.

**Comment by pszlaga** [comment]
Additionally, I removed the network card drivers and installed them again, but this did not solve the problem. I uninstalled VPN, VirtualBox, same problem.

**Comment by keith-horton** [comment]
Wow. sorry it's been so painful. The traces suggest something may be crashing. could you follow this to see if there's something crashing - and collect the dmp?

https://github.com/microsoft/WSL/blob/master/CONTRIBUTING.md#11-reporting-a-wsl-process-crash

**Comment by pszlaga** [comment]
I followed step 11 step by step, but my crashes folder is empty?

**Comment by agxpro** [comment]
I'm facing the same issue after updating Windows 11 to KB5030310 (OS Build 22621.2361) Preview. None of the virtualization apps are working including WSL2, Windows Sandbox, VMware.

**Comment by keith-horton** [comment]
Sorry, I don't know what is causing vswitch to immediately fail to create a switch. All I can recommend at this point is to call to Windows OS support :(

**Comment by afifafifafifafifali** [comment]
Try starting WSL in Safe mode. Image
![Uploading Screenshot (209).png…]()

**Comment by afifafifafifafifali** [comment]
Go to WSL settings app. Go to optional settings and enable safe mode.
Hope this helps.

**Comment by congcnit** [comment]
I wanted to run Docker on my Windows machine but encountered a similar issue.
After a few attempts, I discovered that I couldn’t install distributions in version 2, but with version 1, it was fine.
After successfully installing the distribution and launching Docker Desktop, I still faced a network error. However, after following the recommendation from Docker Desktop’s notification to add the `networkingMode=None` option to the `.wslconfig` file, I am now able to use Docker Desktop.
I hope this information might be helpful to you or others in a similar situation.
![image](https://github.com/user-attachments/assets/b246ec09-2210-43f9-946d-8a741b2298a6)
`C:\Users\cong.nguyen\.wslconfig`
![image](https://github.com/user-attachments/assets/04012429-17db-4729-b9cc-ef34174a5f5b)

**Comment by afifafifafifafifali** [comment]
> Go to WSL settings app. Go to optional settings and enable safe mode. Hope this helps.

Download the wsl from github.
